api/class-wc-rest-dev-orders-controller.php

```diff
@@ -28,4 +28,51 @@ class WC_REST_Dev_Orders_Controller extends WC_REST_Orders_Controller {
 	 */
 	protected $namespace = 'wc/v3';
 
+	/**
+	 * Prepare objects query.
+	 *
+	 * @since  3.0.0
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_objects_query( $request ) {
+		global $wpdb;
+
+		// This is a hack to get around WC_REST_Orders_Controller::prepare_objects_query
```

Comment:
I don't think filtering will help here, the $request is still sent with the status array causing the array to string notice. I've updated the comment to be a bit clearer about _why_ it's unsetting status.

If we can update the core endpoint whenever this is merged, that would be great, but since we probably want this in by the calypso rollout on thursday, this seems best for now.